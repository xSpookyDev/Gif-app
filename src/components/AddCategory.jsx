import PropTypes from 'prop-types'
import { useState } from 'react'


export const AddCategory = ({onNewCategory}) => {

    const [inputValue, setInputValue] = useState("")
    const onInputChange = ({target})=>{
        setInputValue(target.value)
    }
    const onSubmit = (event)=>{
        event.preventDefault();
        if(inputValue.trim().length <=1 ) return;
        onNewCategory(inputValue.trim());
        setInputValue('')
    }
  return (
    <>
    <form action="" onSubmit={(event)=> onSubmit(event)} aria-label="form" role="form">
    <input
    type="text"
    placeholder="Search gif" 
    value={inputValue}
    onChange={onInputChange }/>
    </form>
    </>
  )
}

AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired


}