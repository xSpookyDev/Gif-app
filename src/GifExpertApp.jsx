import { useState} from "react";
import { AddCategory, GifGrid } from "./components";

export const GifExpertApp = () => {
  const [categories, setCategories] = useState(["one punch"]);

  const OnAddCategory = (newCategory) => {
    if (categories.includes(newCategory)) return;
    setCategories([newCategory, ...categories]);
  };

  return (
    <>
      <h1>GifExpertApp</h1>
      <AddCategory onNewCategory={OnAddCategory} />
      {categories.map((category) => {
        return <GifGrid key={category} category={category} />;
      })}
    </>
  );
};
