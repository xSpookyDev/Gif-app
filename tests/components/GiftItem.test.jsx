import { render, screen } from "@testing-library/react"
import { GiftItem } from "../../src/components"

const title = 'Un titulo';
const url = 'https://localhost/cualquier/cosa.jpg';

describe('Prueba en <GifsItems/>', () => {

   
    test('Debe de hacer match con el snapshot', ()=>{
       
        const {container}=  render(<GiftItem title={title} url={url}/>);
       expect(container).toMatchSnapshot();
    })
  test('Debe de mostrar el titulo en el componente', () => {
                render(<GiftItem title={title} url={url}/>);
                expect( screen.getByText(title)).toBeTruthy();
            })   

})


     test('Debe de mostrar la imagen con el URL y el ALT indicado', () => {

            render(<GiftItem title={title} url={url}/>);
            //screen.debug()
            //expect(screen.getByRole('img').src).toBe(url);
          //  expect(screen.getByRole('img').alt).toBe(title);
            const {src, alt} = screen.getByRole('img');
            expect(src).toBe(url);
            expect(alt).toBe(title);


          
     })