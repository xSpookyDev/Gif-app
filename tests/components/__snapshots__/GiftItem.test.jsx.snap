// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Prueba en <GifsItems/> Debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Un titulo"
      src="https://localhost/cualquier/cosa.jpg"
    />
    <p>
      Un titulo
    </p>
  </div>
</div>
`;
